perma_web/tasks/dev.py

```diff
@@ -1110,3 +1114,78 @@ def email_retained_users(ctx, reports_dir='.'):
     logger.info(f"Emailed {sent_count} users")
     if failed_list:
         logger.warning(f"Some users were not emailed: {str(failed_list)}. Check log for fatal SMTP errors.")
+
+
+# WACZ CONVERSION
+
+def save_warc_for_conversion(warc, warcs_dir, file_name):
+    """
+    Gets the file path
+    Get the associates file's contents from storage
+    Saves the file
+    """
+    file_path = os.path.join(f"{warcs_dir}", warc)
+    contents = default_storage.open(file_name).read()
+    with open(file_path, 'wb') as new_f:
+        new_f.write(contents)
+
+
+def get_warcs_for_conversion(source_csv=None, single_warc=None):
+    """
+    Gets the sample WARC ids from the CSV file or constructs from single_warc argument
+    Retrieves the associated file's contents from storage
+    Downloads them to a temp folder
+    """
+    warcs_dir = tempfile.mkdtemp()
+
+    if source_csv:
+        sample_data_guids = os.path.abspath(source_csv)
+        with open(sample_data_guids, mode='r') as file:
+            csv_file = csv.reader(file)
+            for line in csv_file:
+                save_warc_for_conversion(line[1], warcs_dir, f"{line[2]}/{line[1]}")
+    else:
+        # needed to do a bit of reverse engineering here to construct the warc storage file path
```

Comment:
If we were operating on Link objects, which we might well do in practice, we could use the Link's `guid_as_path()` method:
https://github.com/harvard-lil/perma/blob/7f328c8e2e57bdab12fc08fd112a6c63ab15e858/perma_web/perma/models.py#L1667-L1671